dist/fedora-infra/Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM fedora:30
+
+ADD . /src
+
+# build: system utilities and libraries
+RUN dnf -y install g++ openssl-devel
+
+# build: pinned Rust stable toolchain (only use in case of temporary distro breakage)
+#ADD https://static.rust-lang.org/dist/rust-1.35.0-x86_64-unknown-linux-gnu.tar.gz rust.tar.gz
+#RUN tar -xf rust.tar.gz --strip 1 && ./install.sh
+
+# build: system Rust toolchain
+RUN dnf -y install rust cargo
```

Comment:
you might want to pin this by version to make this Dockerfile reproducible

Reply:
This has (on purpose) some room to float. Version gating is done by whatever Fedora decides to promote to stable/F30. This allows fedora-infra to rebuild all the images with fixed packages whenever they need.

Reply:
I see. That approach leaves the risk of re-using an old version until the cache is pruned. I'm leaning towards the idealistic side here as reproducibility issues are easy to miss when debugging. I don't want to block this PR on this discussion though as the scope is much larger.

Reply:
Good point. I'm taking a note to double-check and bypass any caching at the build site.